snet_cli/mpe_client_command.py

```diff
@@ -0,0 +1,186 @@
+        channel_dir = self.get_channel_dir()
+        pfiles = [str(os.path.basename(p)) for p in channel_dir.glob("*.py")]
+        if (len(pfiles) != 2):
+            self._error("We should have exactly two .py files in %s\n"%channel_dir +
+                        "You should remove %s/.py and run _compile_from_file again"%channel_dir)
+
+        # normally we should use importlib for import (see https://stackoverflow.com/questions/67631/how-to-import-a-module-given-the-full-path)
+        # but here we cannot do it because <SERVICE>_pb2_grpc.py import <SERVICE>_pb2.py ... 
+        # so we were forced to play with sys.path
+        sys.path.append(str(channel_dir))
+        
+        # make import 
+        for p in pfiles:
+            
+            to_import = p.replace(".py","")
+            # we check that we cannot be attacked via exec (check to_import)
+            self._check_isidentifier(to_import)
+            exec("import %s"%to_import, globals())
+            
+        service_name = pfiles[0].split("_")[0]   
+        
+        # we check that we cannot be attacked via eval (check service_name)
+        self._check_isidentifier(service_name)
+        stub_class = eval("%s_pb2_grpc.%sStub"%(service_name, service_name))
+        
+        service_descriptor = eval("%s_pb2.DESCRIPTOR.services_by_name['%s']"%(service_name, service_name))
+        is_found = False
+        for method in service_descriptor.methods:
+            if(method.name == self.args.method):
+                request_name  = method.input_type.name
+                response_name = method.output_type.name
+                is_found = True
+                
+        if (not is_found):
+            self._error("Cannot find method %s in the protobuf"%(self.args.method))
+        
+        self._check_isidentifier(request_name)  # it is an overkill, but let's check it also
+        request_class = eval("%s_pb2.%s"%(service_name, request_name))
+
+        return stub_class, request_class
+
+    def _check_isidentifier(self, s):
+        if (not s.isidentifier()):
+            self._error('"%s" is not an identifier'%s)
+        
+    def call_server(self):
+        params  = self._get_call_params()
+        stub_class, request_class = self._import_protobuf_for_channel()
+        request = request_class(**params)
+        channel = grpc.insecure_channel(self.args.endpoint)
+        
+        stub    = stub_class(channel)
+        call_fn = getattr(stub, self.args.method)
+        
+        signature = self._sign_message()
+        metadata = [("snet-payment-type",                 "escrow"                    ),
+                    ("snet-payment-channel-id",            str(self.args.channel_id)  ), 
+                    ("snet-payment-channel-nonce",         str(self.args.nonce)       ), 
+                    ("snet-payment-channel-amount",        str(self.args.amount)      ),
+                    ("snet-payment-channel-signature-bin", base64.b64encode(signature))]
```

Comment:
As far as I understand specification you don't need encode it in base64 - gRPC library will do it for you.